pvlib/iam.py

```diff
@@ -274,6 +274,119 @@ def martin_ruiz(aoi, a_r=0.16):
+    ground-reflected irradiance using the Martin and Ruiz incident angle model.
+
+    Parameters
+    ----------
+    surface_tilt: float or array-like, default 0
+        Surface tilt angles in decimal degrees.
+        The tilt angle is defined as degrees from horizontal
+        (e.g. surface facing up = 0, surface facing horizon = 90)
+
+    a_r : numeric
+        The angular losses coefficient described in equation 3 of [1].
+        This is an empirical dimensionless parameter. Values of a_r are
+        generally on the order of 0.08 to 0.25 for flat-plate PV modules.
+        a_r must be greater than zero.
+
+    c1 : numeric scalar
+        First fitting parameter for the expressions that approximates the
+        integral of diffuse irradiance coming from different directions.
+        c1 is given as the constant 4 / 3 / pi (0.4244) in [1].
+
+    c2 : numeric scalar
+        Second fitting parameter for the expressions that approximates the
+        integral of diffuse irradiance coming from different directions.
+        If c2 is None, it will be calculated according to the linear
+        relationship given in [3].
+
+    Returns
+    -------
+    iam_sky : numeric
+        The incident angle modifier for sky diffuse
+
+    iam_ground : numeric
+        The incident angle modifier for ground-reflected diffuse
+
+    Notes
+    -----
+    Sky and ground modifiers are complementary: iam_sky for tilt = 30 is
+    equal to iam_ground for tilt = 180 - 30.  For vertical surfaces,
+    tilt = 90, the two factors are equal.
+
+    References
+    ----------
+    [1] N. Martin and J. M. Ruiz, "Calculation of the PV modules angular
+    losses under field conditions by means of an analytical model", Solar
+    Energy Materials & Solar Cells, vol. 70, pp. 25-38, 2001.
+
+    [2] N. Martin and J. M. Ruiz, "Corrigendum to 'Calculation of the PV
+    modules angular losses under field conditions by means of an
+    analytical model'", Solar Energy Materials & Solar Cells, vol. 110,
+    pp. 154, 2013.
+
+    [3] "IEC 61853-3 Photovoltaic (PV) module performance testing and energy
+    rating - Part 3: Energy rating of PV modules". IEC, Geneva, 2018.
+
+    See Also
+    --------
+    iam.martin_ruiz
+    iam.physical
+    iam.ashrae
+    iam.interp
+    iam.sapm
+    '''
+    # Contributed by Anton Driesse (@adriesse), PV Performance Labs. Oct. 2019
+
+    if isinstance(surface_tilt, pd.Series):
+        out_index = surface_tilt.index
+    else:
+        out_index = None
+
+    surface_tilt = np.asanyarray(surface_tilt)
+
+    # undo possible surface rotations
+    with np.errstate(invalid='ignore'):
+        surface_tilt = (surface_tilt + 180) % 360 - 180
+
+    # flip backward tilts forward
+    surface_tilt = np.abs(surface_tilt)
+
+    # avoid undefined results for horizontal or upside-down surfaces
+    small_angle = 1e-06
+    surface_tilt = np.clip(surface_tilt, small_angle, 180 - small_angle)
```

Comment:
This line is limiting `surface_tilt` between (nearly) 0 and 180, correct?

Reply:
Yes, but it only works properly when the angles are reflected/rotated into the range 0-180 beforehand.  Otherwise it would force -30 to nearly 0 and 210 to nearly 180, etc.